syncapi/streams/stream_pdu.go

```diff
@@ -2,18 +2,48 @@ package streams
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	"github.com/matrix-org/dendrite/syncapi/types"
 	userapi "github.com/matrix-org/dendrite/userapi/api"
 	"github.com/matrix-org/gomatrixserverlib"
+	"go.uber.org/atomic"
 )
 
+const PDU_STREAM_QUEUESIZE = 2048
+const PDU_STREAM_WORKERS = 256
```

Comment:
Please comment what these configure and why too low is bad and too high is bad e.g:
```go
// The max number of per-room goroutines to have running.
// Too high and this will consume lots of CPU, too low and sync responses will take longer to process.
const PDU_STREAM_WORKERS = 256
```